packages/@yoda/flora/index.js

```diff
@@ -2,259 +2,161 @@
-      this.callbacks[0](args[0], args[1], args[2])
-      break
-    // disconnected
-    case 1:
-      this.callbacks[1]()
-      break
-  }
-}
 
 /**
- * @memberof module:@yoda/flora
- * @member {Number} MSGTYPE_INSTANT
+ * @memberof module:@yoda/flora~Response
+ * @member {Number} retCode
  */
-exports.MSGTYPE_INSTANT = 0
+
 /**
- * @memberof module:@yoda/flora
- * @member {Number} MSGTYPE_PERSIST
+ * @memberof module:@yoda/flora~Response
+ * @member {Array} msg
  */
-exports.MSGTYPE_PERSIST = 1
-// exports.MSGTYPE_REQUEST = 2;
 
-exports.CLI_SUCCESS = 0
-// 认证失败
-exports.CLI_EAUTH = -1
-// 调用参数错误
-exports.CLI_EINVAL = -2
-// 连接断开
-exports.CLI_ECONN = -3
-// 'get'请求超时
-// exports.CLI_ETIMEOUT = -4;
-// 'get'请求目标不存在
-// exports.CLI_ENEXISTS = -5;
+/**
+ * @memberof module:@yoda/flora~Response
+ * @member {String} sender
+ */
 
 /**
- * @class
- * @classdesc utility of data struct serialize
+ * @callback module:@yoda/flora~SubscribeCallback
+ * @param {Array} - msg content
+ * @param {Number} - type of msg
+ * @returns {module:@yoda/flora~Reply} reply message to sender of this REQUEST message
  */
-function Caps () {
-  this.pairs = []
-}
 
-Caps.INT32 = 105 // 'i'
-Caps.INT64 = 108 // 'l'
-Caps.FLOAT = 102 // 'f'
-Caps.DOUBLE = 100 // 'd'
-Caps.STRING = 83 // 'S'
-Caps.BINARY = 66 // 'B'
-Caps.OBJECT = 79 // 'O'
+var Agent = require('./flora-cli.node').Agent
 
 /**
- * write int32 value
- * @memberof module:@yoda/flora~Caps
- * @method writeInt32
- * @param {Number} i32 - int32 value
- */
-Caps.prototype.writeInt32 = function (i32) {
-  if (typeof i32 !== 'number') { return }
-  var p = { type: Caps.INT32, value: i32 }
-  this.pairs.push(p)
+ * subscribe flora msg
+ * @method subscribe
+ * @memberof module:@yoda/flora~Agent
+ * @param {String} name - msg name for subscribe
+ * @param {module:@yoda/flora~SubscribeCallback} cb - callback if received msg that subscribed
+ */
+Agent.prototype.subscribe = function (name, cb) {
+  this.nativeSubscribe(name, (msg, type) => {
+    try {
+      return cb(msg, type)
+    } catch (e) {
+      process.nextTick(() => {
+        throw e
+      })
+    }
+  })
 }
 
 /**
- * write int64 value
- * @memberof module:@yoda/flora~Caps
- * @method writeInt64
- * @param {Number} i64 - int64 value
+ * post msg and get response
+ * @method get
+ * @memberof module:@yoda/flora~Agent
+ * @param {String} name - msg name
+ * @param {Array} [msg] - msg content
+ * @returns {Promise} promise that resolves with an array of {module:@yoda/flora~Response}
  */
-Caps.prototype.writeInt64 = function (i64) {
-  if (typeof i64 !== 'number') { return }
-  var p = { type: Caps.INT64, value: i64 }
-  this.pairs.push(p)
+Agent.prototype.get = function (name, msg) {
+  if (typeof name !== 'string') {
+    return Promise.reject(exports.ERROR_INVALID_PARAM)
+  }
+  if (msg !== undefined && msg !== null && !Array.isArray(msg)) {
+    return Promise.reject(exports.ERROR_INVALID_PARAM)
+  }
+  var ret = new Promise((resolve, reject) => {
```

Comment:
```suggestion
  return new Promise((resolve, reject) => {
```

No necessary to new this variable :)